controllers/spacecleanup/space_cleanup_controller.go

```diff
@@ -97,6 +98,8 @@ func (r *Reconciler) ensureDeletionIfNeeded(logger logr.Logger, space *toolchain
 		if err := r.Client.Delete(context.TODO(), space); err != nil {
 			return false, 0, errs.Wrap(err, "unable to delete Space")
 		}
+
+		counter.DecrementSpaceCount(logger, space.Spec.TargetCluster)
```

Comment:
Ok, I did both of them because at some point in the many E2E tests I've run locally there were some leftovers in the metrics values. When eh `captureBaselineValues` were printed the SpaceCount was starting from 15 instead o 0. But I can try and remove it and see if everything is still passing.